src/internal/connector/graph/betasdk/beta_client.go

```diff
@@ -0,0 +1,86 @@
+//
+// Changes to Sites Directory:
+// Access files send requests with an adapter's with ASync() support.
+// This feature is not enabled in v1.0. Manually changed in remaining files.
+// Additionally, only calls that begin as client.SitesBy(siteID).Pages() have an endpoint.
+//
+// The use case specific to Pages(). All other requests should be routed to the /internal/connector/graph.Servicer
+// Specifics on `betaClient.SitesById(siteID).Pages` are located: sites/site_item_request_builder.go
+//
+// Required model files are identified as `modelFiles` in kiota-lock.json. Directory -> betasdk/models
+// Required access files are identified as `sitesFiles` in kiota-lock.json. Directory -> betasdk/sites
+//
+// BetaClient minimal msgraph-beta-sdk-go for connecting to msgraph-beta-sdk-go
+// for retrieving `SharePoint.Pages`. Code is generated from kiota.dev.
+// requestAdapter is registered with the following the serializers:
+// --  "Microsoft.Kiota.Serialization.Json.JsonParseNodeFactory",
+// --  "Microsoft.Kiota.Serialization.Text.TextParseNodeFactory",
+// --  "Microsoft.Kiota.Serialization.Form.FormParseNodeFactory"
+type BetaClient struct {
+	// Path parameters for the request
+	pathParameters map[string]string
+	// The request adapter to use to execute the requests.
+	requestAdapter *msgraphsdk.GraphRequestAdapter
+	// Url template to use to build the URL for the current request builder
+	urlTemplate string
+}
+
+// NewBetaClient instantiates a new BetaClient and sets the default values.
+// func NewBetaClient(requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*BetaClient) {
+func NewBetaClient(requestAdapter *msgraphsdk.GraphRequestAdapter) *BetaClient {
+	m := &BetaClient{}
+	m.pathParameters = make(map[string]string)
+	m.urlTemplate = "{+baseurl}"
+	m.requestAdapter = requestAdapter
+	absser.RegisterDefaultSerializer(func() kioser.SerializationWriterFactory {
+		return kw.NewJsonSerializationWriterFactory()
+	})
+	absser.RegisterDefaultSerializer(func() kioser.SerializationWriterFactory {
+		return ktext.NewTextSerializationWriterFactory()
+	})
+	absser.RegisterDefaultSerializer(func() kioser.SerializationWriterFactory {
+		return kform.NewFormSerializationWriterFactory()
+	})
+	absser.RegisterDefaultDeserializer(func() kioser.ParseNodeFactory {
+		return kw.NewJsonParseNodeFactory()
+	})
+	absser.RegisterDefaultDeserializer(func() kioser.ParseNodeFactory {
+		return ktext.NewTextParseNodeFactory()
+	})
+	absser.RegisterDefaultDeserializer(func() kioser.ParseNodeFactory {
+		return kform.NewFormParseNodeFactory()
+	})
+
+	if m.requestAdapter.GetBaseUrl() == "" {
+		m.requestAdapter.SetBaseUrl("https://graph.microsoft.com/beta")
+	}
+	return m
+}
+
+// SitesById provides operations to manage the collection of site entities.
+func (m *BetaClient) SitesById(id string) *i1a3c1a5501c5e41b7fd169f2d4c768dce9b096ac28fb5431bf02afcc57295411.SiteItemRequestBuilder {
```

Comment:
What is the ID in this case?  Also, the name `SitesById` makes it sound like the one ID returns many sites.  Is that true?

Reply:
The Kiota naming conventions favor plural notation when generated. We have implemented this code in other areas:
```go
	gc.Service.Client().UsersById()
	gc.Service.Client().UsersById().MessagesById()
```